### PR TITLE
Added documentation for 3D Joints, the 2D Pin Joint, and the Physicsservers, also minor Bugfix

### PR DESCRIPTION
**Pull request by Grosskopf**
Bringing Documentation status a +4% to overall completion this PR is quite big, a lot of it is copy&paste work though :)

### CODE REVIEW

File: doc/classes/ConeTwistJoint.xml [hunk 1]
```diff
@@ -30,28 +32,39 @@
...
+			The strength with which the correction of an overstepped swing limit will take place, the higher, the stronger.
 		</member>
 		<member name="relaxation" type="float" setter="set_param" getter="get_param">
+			A factor that gets applied to the speed of all twist and swing impulses.
 		</member>
 		<member name="softness" type="float" setter="set_param" getter="get_param">
+			The ease with which the joint starts to twist, if it's too low, it takes more force to start twisting the joint.
 		</member>
 		<member name="swing_span" type="float" setter="_set_swing_span" getter="_get_swing_span">
+			The higher, the more swing (rotation around axes orthogonal to the twist axis) is needed to have the swing corrected. Disabled when under 0.05.
```
Comment (NathanLovato) [inline doc/classes/ConeTwistJoint.xml]: Avoid parens. Avoid passive voice. Favor action verbs and direct voice. Example for the form (check it's correct, a lot is guesswork, I'm not familiar with the physics engine and joints in godot):
```Controls the frequency of swing corrections by the physics engine.
Swing is rotation from side to side, around the axis perpendicular to the twist axis. The lower the value, the more often the physics engine will smooth out the swing. 
A value lower than [code]0.05[/code] disables all swing corrections. Default value: [code]value[/code]```

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Avoid parens. Avoid passive voice. Favor action verbs and direct voice.
There's a *lot* of passive voice in the PR

File: doc/classes/ConeTwistJoint.xml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="ConeTwistJoint" inherits="Joint" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		A twist joint between two 3D bodies
 	</brief_description>
 	<description>
+		The joint can be rotated across the axis in the middle of the x-axes of the bodies.
```
[the remark this answers is not in the document]
Comment (Grosskopf) [reply doc/classes/ConeTwistJoint.xml]: i'll research that more, there are those pivot points at each body, and it takes the middle out of the two x axes of those pivot points, i'm not 100% shure how it generates them though...

File: doc/classes/ConeTwistJoint.xml [hunk 3]
```diff
@@ -30,28 +32,39 @@
 	</methods>
 	<members>
 		<member name="bias" type="float" setter="set_param" getter="get_param">
+			The strength with which the correction of an overstepped swing limit will take place, the higher, the stronger.
 		</member>
 		<member name="relaxation" type="float" setter="set_param" getter="get_param">
+			A factor that gets applied to the speed of all twist and swing impulses.
```
[the remark this answers is not in the document]
Comment (Grosskopf) [reply doc/classes/ConeTwistJoint.xml]: i hope it is in the ]0;1] range, but i'm unsure where to search for that info

File: doc/classes/ConeTwistJoint.xml (hunk 1 above)
Comment (Grosskopf) [reply doc/classes/ConeTwistJoint.xml]: thanks, relatively precisely what i think it does, i should definitely not be so shy to write multiple lines of explanation :)

**Comment by Grosskopf** [comment]
i think this should also be added to bug since it changes something in the sourcecode @Hinsbart :)

**Comment by Grosskopf** [comment]
i think this should also be added to bug since it changes something in the sourcecode @Hinsbart :)

File: doc/classes/ConeTwistJoint.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Grosskopf) [reply doc/classes/ConeTwistJoint.xml]: *_* i found it, now a lot has gotten more clear to me, now i only need to understand what the affine inverse transformation-matrix of the body A multiplied with the transformationmatrix of the joint-object means in human-language :D

[the remark this answers is not in the document]
Comment (Grosskopf) [reply doc/classes/ConeTwistJoint.xml]: ooooh i now understand a looot more i think :)

File: doc/classes/ConeTwistJoint.xml [hunk 4]
```diff
@@ -30,28 +32,39 @@
 	</methods>
 	<members>
 		<member name="bias" type="float" setter="set_param" getter="get_param">
+			The strength with which the correction of an overstepped swing limit will take place, the higher, the stronger.
```
[the remark this answers is not in the document]
Comment (Grosskopf) [reply doc/classes/ConeTwistJoint.xml]: it defines how fast correction of the limits take place, just noticed it gets multiplied to twist as well, amount of time until everything is in limits without relaxation force is 1/bias seconds from what i can see in the code...

**Review by NathanLovato — APPROVED** [review]

**Comment by NathanLovato** [comment]
@Hinsbart should the bug fix be split into another PR? Just another commit? Or is it fine this way, one commit for docs + bug fix?

This is already a great PR. It fixes a huge gap in Physics and as far as I'm concerned it's ready to merge.
With so many descriptions, there are always things one could improve here and there, but it's a classes few of us understand, and we'll need feedback from actual users to refine the docs eventually.

**Comment by Grosskopf** [comment]
I know it's a bit fishy, doing a bugfix with the documentation together, and others will have the final say on this topic but as an argument for doing this: the functions that were influenced are not used internally at the moment and if the bugfix gets merged or not influences the corresponding lines in the documentation just as much as i believe the documentation getting merged influences the necessity of the bugfix

**Comment by NathanLovato** [comment]
@akien-mga This PR adds a lot of entries to the classref and fixes a bug at the same time. The bugfix affects the docs and vice-versa. It's all in one commit. The content is good - can we merge as is or should the commits/PRs be split?

**Comment by Hinsbart** [comment]
I'd say it's fine in this case.

**Pull request merged**